docs/content/dev/notes.md

```diff
@@ -0,0 +1,55 @@
+# Notes
+
+Each note in HedgeDoc 2 contains the following information:
+
+- publicId (`b604x5885k9k01bq7tsmawvnp0`)
+- a list of aliases (`[hedgedoc-2, hedgedoc-next]`)
+- groupPermissions
+- userPermissions
+- viewCount (`0`)
+- owner
+- revisions
+- authorColors
+- historyEntries
+- description (`All you never wanted to know about notes`)
+- title (`Notes`)
+- tags (`[features, cool, update]`)
+- version
+
+The `publicId` is the default possibility of identifying a note. It will be a randomly generated 128-bit value encoded with [base32-encode](https://www.npmjs.com/package/base32-encode) using the crockford variant and converted to lowercase. We use lowercase, because we want to minimize case confusion.
+
+`aliases` are the other way of identifying a note. There can be any number of them, and the owner of the note is able to add or remove them. All aliases are just strings (especially to accommodate the old identifier from HedgeDoc 1 [see below](#conversion-of-hedgedoc-1-notes)), but new aliases added with HedgeDoc 2 will only allow the following characters `[a-z0-9\-_]`. This is done to once again prevent case confusion.
+
+`groupPermissions` and `userPermissions` each hold a list of the appropriate permissions. Each of those holds a reference to a note and a user/group and specify what the user/group is allowed to do. Each of the permissions is additive, that means a user that has only the right to read a note via a group, but the right to write via a different group or directly for his user, is able to write in the note.
+
+The `viewCount` is a simple counter that holds how often the note in question was requested.
+
+`owner` is the user that created the note or later got ownership of the note. The current owner is able to change the owner of the note to someone else.
+The `revisions` hold all revisions of the note. These are the changes to the note content and by which they were performed.
+
+The `authorColors` each specify for the tuple user and note which color should be used to highlight them.
+
+The `historyEntries` hold the history entries this note is referenced in.They are mainly here for the purpose of deleting the history entries on note deletion.
+
+`description`, `tags` and `title` are each information specified in the frontmatter of the note. They are extracted and saved in the database to allow the history page to show them and do a search for tags without having to do a full-text search or having to parse the tags of each note on search.
```

Comment:
The term "frontmatter" should be defined.

Reply:
not to sure where to put that. Maybe someone else has a good idea?

Reply:
We could just link to an external definition, like Jekyll's.

Reply:
You mean https://jekyllrb.com/docs/front-matter/?

Reply:
Maybe also refer to our documentation about allowed frontmatter fields?

Reply:
I added that